comp/core/config/config.go

```diff
@@ -29,18 +30,24 @@ type cfg struct {
 
 type configDependencies interface {
 	getParams() *Params
+	getSecretResolver() secrets.Component
 }
 
 type dependencies struct {
 	fx.In
 
 	Params Params
+	Secret secrets.Component
 }
 
 func (d dependencies) getParams() *Params {
 	return &d.Params
 }
 
+func (d dependencies) getSecretResolver() secrets.Component {
+	return d.Secret
+}
+
```

Comment:
I don't get why the `config` comp expose the secrets comp ?

Reply:
Config depends on secrets. Here https://github.com/DataDog/datadog-agent/pull/20883/files#diff-129fdbc93cd441b765523713ca5c79e3e10a2aa1e736b5c6bf37784a508af645R55 the config gets the secrets component in order to load and decrypt the config.

Reply:
Or are you just pointing out that it shouldn't be an exported field?